pytorch_lightning/utilities/data.py

```diff
@@ -69,6 +67,13 @@ def extract_batch_size(batch: BType) -> int:
             )
             break
 
+    if batch_size is None:
+        raise MisconfigurationException(
+            "Trying to infer `batch_size` from an ambiguous collection but couldn't find one. Either"
+            " set `self.log(..., batch_size=batch_size)` or simplify the batch collection passed"
+            " inside `*_step` functions."
+        )
```

Comment:
```suggestion
        raise MisconfigurationException(
            "Trying to infer `batch_size` from an ambiguous collection but couldn't find one. Either"
            " set `self.log(..., batch_size=batch_size)`.
        )
```

Reply:
Let's not introduce a new API here. It should be an int.

Reply:
We could use `-1` for that if you want to stick with ints? Although I think `None` to be more intuitive

Reply:
Hey @rohitgr7,

As you said, this would raise an Exception. I just wanted to share that I don't understand what you meant to say here

>  Or simplify the batch collection passed " inside `*_step` functions.

To me, if the batch_size can't be inferred properly, then the user should pass the `batch_size`, not the batch.

Reply:
there are two options. Either they can pass the batch_size to self.log(), or they can simplify the batch being passed to step functions. I have added both suggestions in the error msg.

Reply:
Oh I understand what you meant now.

> `simplify the batch collection passed` is quite confusing.
Maybe something like. Lightning couldn't infer the `batch_size` from your batch. Either simplify its structure or provide the batch_size as self.log(..., batch_size=batch_size) `

Reply:
better! updated